### PR TITLE
Fix warnings in pegasus and runtime in interactive engine 

### PR DESCRIPTION
**Pull request by BingqingLyu**
## Related issue number

#593

### CODE REVIEW

File: interactive_engine/src/executor/Pegasus/src/execute.rs [hunk 1]
```diff
@@ -484,7 +484,11 @@ impl<T: Task> ThreadPool<T> {
                             ::std::thread::park();
                             running = in_running.load(Ordering::SeqCst);
                         }
-                        let try_swap = in_running.compare_and_swap(running, running + 1, Ordering::SeqCst);
+                        let try_swap = match in_running.compare_exchange(running, running + 1, Ordering::SeqCst, Ordering::SeqCst) {
+                            Ok(x) => x,
+                            Err(x) => x,
+                        };
+                   //     let try_swap = in_running.compare_and_swap(running, running + 1, Ordering::SeqCst);
```
Comment (tianliplus) [inline interactive_engine/src/executor/Pegasus/src/execute.rs:491]: remove unneccessary comment

File: interactive_engine/src/executor/Pegasus/src/execute.rs [hunk 2]
```diff
@@ -529,7 +533,11 @@ impl Executor<GenericTask> for ThreadPool<GenericTask> {
             Err(RejectError(task))
         } else {
             loop {
-                let new_running = self.in_running.compare_and_swap(running, running + 1, Ordering::SeqCst);
+                let new_running = match self.in_running.compare_exchange(running, running + 1,  Ordering::SeqCst, Ordering::SeqCst) {
+                    Ok(x) => x,
+                    Err(x) => x,
+                } ;
+            //    let new_running = self.in_running.compare_and_swap(running, running + 1, Ordering::SeqCst);
```
Comment (tianliplus) [inline interactive_engine/src/executor/Pegasus/src/execute.rs:540]: same as above

**Review by tianliplus — COMMENTED** [review]
Remove the comments

File: interactive_engine/src/executor/Pegasus/src/execute.rs (hunk 1 above)
Comment (BingqingLyu) [reply interactive_engine/src/executor/Pegasus/src/execute.rs]: Done

File: interactive_engine/src/executor/Pegasus/src/execute.rs (hunk 2 above)
Comment (BingqingLyu) [reply interactive_engine/src/executor/Pegasus/src/execute.rs]: Done

**Review by tianliplus — APPROVED** [review]

**Pull request merged**